packages/contracts/contracts/factory/PluginRepoFactory.sol

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.8.10;
+
+import "../utils/Proxy.sol";
+import "../registry/AragonPluginRegistry.sol";
+import "../APM/PluginRepo.sol";
+
+/// @title PluginRepoFactory to create a PluginRepo
+/// @author Sarkawt Noori - Aragon Association - 2022
+/// @notice This contract is used to create a PluginRepo and register it on AragonPluginRegistry contract.
+contract PluginRepoFactory {
+    AragonPluginRegistry public aragonPluginRegistry;
+    address public pluginRepoBase;
+
+    error ApmRegEmptyName();
+
+    constructor(AragonPluginRegistry _aragonPluginRegistry) {
+        aragonPluginRegistry = _aragonPluginRegistry;
+
+        setupBases();
+    }
+
+    /// @notice Create new pluginRepo in registry with `_name`
+    /// @param _name PluginRepo name, must be ununsed
+    /// TODO: Rethink if it need permission to prevent it from getting poluted, same for newPluginRepoWithVersion
+    function newPluginRepo(string calldata _name, address _dev) external returns (PluginRepo) {
+        return _newPluginRepo(_name, _dev);
+    }
+
+    /// @notice Create new pluginRepo in registry with `_name` and publish a first version with contract `_pluginFactoryAddress` and content `@fromHex(_contentURI)`
+    /// @param _name PluginRepo name
+    /// @param _initialSemanticVersion Semantic version for new pluginRepo version
+    /// @param _pluginFactoryAddress address for smart contract logic for version (if set to 0, it uses last versions' contractAddress)
+    /// @param _contentURI External URI for fetching new version's content
+    function newPluginRepoWithVersion(
+        string calldata _name,
+        uint16[3] memory _initialSemanticVersion,
+        address _pluginFactoryAddress,
+        bytes memory _contentURI,
+        address _dev
+    ) public returns (PluginRepo pluginRepo) {
```

Comment:
```suggestion
    ) external returns (PluginRepo pluginRepo) {
```
Is there a reason why this is public and not external? Where is it used internally?

Reply:
thanks fro spiting this.
amended